src/Controller/CRUDController.php

```diff
@@ -212,7 +212,7 @@ public function batchActionDelete(ProxyQueryInterface $query)
      *
      * @return Response|RedirectResponse
      */
-    public function deleteAction($id)
+    public function deleteAction($id = null)
```

Comment:
Adding a default value for an argument in a public method is not BC.

Reply:
Why ?

And this should be considered like a bugfix.
There is some `object/delete?id=1`, failing because an $id param is expected.

Reply:
Because if this method is already extended by the user, the signature will be broken: https://3v4l.org/5rLhM.

Reply:
@phansys But if the method does not work, we should fix it and not wait for the next major for this.

The following code
```
$listMapper->add(
  '_action',
  'actions', [
    'actions' => ['delete'  => []],
  ]
)
```
add a delete button in the admin.

When clicking on it, I get the following error:
```
Could not resolve argument $id of "Sonata\AdminBundle\Controller\CRUDController::deleteaction()", maybe you forgot to register the controller as a service or missed tagging it with the "controller.service_arguments"?
```

Reply:
I think we might be able to fix it by setting a default value directly in the route definition (like 42, for instance). Can you please try that?

Reply:
> I think we might be able to fix it by setting a default value directly in the route definition (like 42, for instance). Can you please try that?

How ? Where ? My route definition are in a yaml file or in a `@Route` annotations, but I didn't found one here. Theres route are generated I-dont-know-how-or-where by the admin.

Reply:
I think it might be here: https://github.com/sonata-project/SonataAdminBundle/blob/f0c22710bed3a5926bab5a65b4c1b73f2e1da671/src/Route/PathInfoBuilder.php#L43 or here: https://github.com/sonata-project/SonataAdminBundle/blob/f0c22710bed3a5926bab5a65b4c1b73f2e1da671/src/Route/QueryStringBuilder.php#L43

Reply:
I tried but it does not solve anything.

Actually, the only solution I found is `$id = null`.
Since the method actually doesn't work, I think the change should be accepted.

Reply:
>  There is some object/delete?id=1, failing because an $id param is expected.

When I read this and see the code in `PathInfoBuilder`, I think there is an issue with the url… shouldn't it be `object/1/delete`? IMO the code that should be fixed is the code that generates the url, not the code which processes it.

Reply:
Ok ! I found the bug !
With `bin/console debug:router` I found others urls with the issue.

![image](https://user-images.githubusercontent.com/9052536/72220588-5356a880-3552-11ea-91a0-ca51d015a8b6.png)
![image](https://user-images.githubusercontent.com/9052536/72220593-65384b80-3552-11ea-94bc-d04fd524411c.png)
![image](https://user-images.githubusercontent.com/9052536/72220595-6bc6c300-3552-11ea-8309-f59959bdc34c.png)

It seems that `$admin->getRouterIdParameter()` return nothing sometimes.

Reply:
@greg0ire If I rename in PathInfoBuilder
![image](https://user-images.githubusercontent.com/9052536/72220739-1c819200-3554-11ea-9781-0cf058e16102.png)

And I rename in QueryStringBuilder
![image](https://user-images.githubusercontent.com/9052536/72220757-4f2b8a80-3554-11ea-97c0-3e311aae42c8.png)

The `debug:router` command give me
![image](https://user-images.githubusercontent.com/9052536/72220758-56eb2f00-3554-11ea-9ada-b580f0ba2a7d.png)
![image](https://user-images.githubusercontent.com/9052536/72220763-623e5a80-3554-11ea-8e1f-e5382cb55335.png)

So every url without `{id}` are coming from somewhere else.
There is a conflict with something else.

Reply:
Ok, i found it ! This is kind of my fault.

In my `ClientAbstractAdmin` I do
![image](https://user-images.githubusercontent.com/9052536/72220822-16d87c00-3555-11ea-9b39-5c7b80e02fd8.png)

Then, in my `ClientAdmin` I do
![image](https://user-images.githubusercontent.com/9052536/72220808-ed1f5500-3554-11ea-86ea-1c8cd66972dc.png)

It's sometimes the same for show, edit, ... in others admin.

I don't think the developer should write:
```
$collection->add('delete', $admin->getRouterIdParameter().'/delete');
```
when he wants to re-add an url he removed.

When re-adding a removed url, it should use the  already registered url.

Reply:
Sounds a bit hard to do, doesn't it? But why not…